lntest/itest/lnd_signer_test.go

```diff
@@ -411,3 +412,75 @@ func deriveCustomizedKey(ctx context.Context, node *lntest.HarnessNode,
 	}
 	return pub, nil
 }
+
+// testSignVerifyMessage makes sure that the SignMessage RPC can be used with
+// all custom ways of specifying by verifying with VerifyMessage. Tests both
+// ecd and schnorr signatures.
+func testSignVerifyMessage(net *lntest.NetworkHarness, t *harnessTest) {
+	runSignVerifyMessage(t, net.Alice)
+}
+
+// runSignVerifyMessage makes sure that the SignMessage RPC can be used with
+// all custom ways of specifying by verifying with VerifyMessage. Tests both
+// ecd and schnorr signatures.
+func runSignVerifyMessage(t *harnessTest, alice *lntest.HarnessNode) {
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	aliceMsg := []byte("alice msg")
+	keyLoc := signrpc.KeyLocator{KeyFamily: 0, KeyIndex: 1}
+
+	// Sign a message with the default ecds.
+	signMsgReq := &signrpc.SignMessageReq{
+		Msg:        aliceMsg,
+		KeyLoc:     &keyLoc,
+		SchnorrSig: false,
+	}
+
+	signMsgResp, err := alice.SignerClient.SignMessage(ctxt, signMsgReq)
+	require.NoError(t.t, err, "failed to sign message")
+
+	customPubKey, err := deriveCustomizedKey(ctxt, alice, &keyLoc)
+	require.NoError(t.t, err, "failed to create customized pubkey")
+
+	verifyReq := &signrpc.VerifyMessageReq{
+		Msg:          aliceMsg,
+		Signature:    signMsgResp.Signature,
+		Pubkey:       customPubKey.SerializeCompressed(),
+		IsSchnorrSig: false}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
```

Comment:
Can reuse the above timeout context, everything here should be done in a few milliseconds anyway. Same for other calls below.